lib/router.js

```diff
@@ -217,6 +217,36 @@ Cherrytree.prototype.destroy = function () {
 }
 
 /**
+ * Check if the given route/params/query combo is active
+ * @param  {String} name   target route name
+ * @param  {Object} params
+ * @param  {Object} query
+ * @return {Boolean}
+ *
+ * @api public
+ */
+Cherrytree.prototype.isActive = function (name, params, query) {
+  params = params || {}
+  query = query || {}
+
+  let activeRoutes = this.state.routes || []
+  let activeParams = this.state.params || {}
+  let activeQuery = this.state.query || []
+
+  let isNameActive = !!activeRoutes.find((route) => {
+    return route.name === name
+  })
+  let areParamsActive = !!Object.keys(params).every((key) => {
+    return activeParams[key] === params[key]
+  })
+  let isQueryActive = !!Object.keys(query).every((key) => {
+    return activeQuery[key] === query[key]
+  })
```

Comment:
could you inline all 3 of these, i.e.:

```js
let isNameActive = !!activeRoutes.find(route => route.name === name)
```

and so on. Note that () are also optional when you only have 1 arg to the arrow function.